### ISSUE TITLE
-sys crates not found

### ISSUE THREAD

**Issue by pentamassiv** [gtk-rs/gir#1346]
TLDR:

I believe I stumbled upon three issues:

- [ ] Ensure that glib, gdk and gtk are present as dependencies if a Cargo.toml file already exists
- [x] Change link for gdk and gtk crates to https://github.com/gtk-rs/gtk3-rs
- [x] Rename the sys packages to the "regular"/safe wraper crate

Hi,
thank you for gir and gtk-rs. Its a great project! I believe I found some minor issues though.

I wanted to update my wrapper for gtk-layer-shell. To make sure everything is up to date because I made some manual edits to generated files last time, I started from scratch again. I followed the tutorial in the book again to create the wrapper for gtk-layer-shell. I get an error when building the sys crate for gtk-layer-shell.

[This](https://github.com/pentamassiv/gtk-layer-shell-gir/commit/6251467178d71721479b92cdb08b61d0fc0b954d) is what the repo looks like before running gir.
If I run `gir -o .` and the old Cargo.toml from creating the new lib crates still exists, I get the following errors:
```
use of undeclared crate or module `glib`
use of undeclared crate or module `gtk`
use of undeclared crate or module `gdk`
```
I guess this is because the old Cargo.toml was present and for some reason the dependencies for gtk, gdk and glib were not added. You can find the resulting code [here](https://github.com/pentamassiv/gtk-layer-shell-gir/commit/a04a17dea6d368cf5542325ae46b86f9f71a04ed).

If I then run `rm Cargo.*` and `gir -o .` for a second time, I get the following error

```
error: no matching package found
searched package name: `gdk-sys`
perhaps you meant:      gio-sys or glib-sys
location searched: https://github.com/gtk-rs/gtk-rs-core
required by package `gtk-layer-shell-sys v0.0.1
```
You can find the resulting code [here](https://github.com/pentamassiv/gtk-layer-shell-gir/commit/6cbfd17093364bfe85c2a8010215a324c6668364).
I assume this is because in the newly created Cargo.toml the dependencies for gtk, gdk and glib were successfully added, but the link for the gdk-sys crate is incorrect. GDK and GTK can actually be found under https://github.com/gtk-rs/gtk3-rs. When I changed the link in the Cargo.toml, I get it to build a bit more. Until I get the next error.

Now I am get the following errors again:
```
use of undeclared crate or module `glib`
use of undeclared crate or module `gtk`
use of undeclared crate or module `gdk`
```
This can be helped if I change my Cargo.toml in the sys crate to:

```
[dependencies.glib]
git = "https://github.com/gtk-rs/gtk-rs-core"
package = "glib-sys"

[dependencies.gdk]
git = "https://github.com/gtk-rs/gtk3-rs"
package = "gdk-sys"

[dependencies.gtk]
git = "https://github.com/gtk-rs/gtk3-rs"
package = "gtk-sys"
```
And finally everything builds just fine :-)

**Comment by sdroege** [comment]
Do you want to add this via a PR to the book?

**Comment by bilelmoussaoui** [comment]
Not only the book but we need to fix the generator for the -sys crates to make it produce the renamed crates as well. Too many people asked about it so far

**Comment by pentamassiv** [comment]
Ultimately I also think correcting the links and renaming the crates would be cleaner. Maybe I can have a look at fixing that next week. I'm guessing https://github.com/gtk-rs/gir/blob/master/src/codegen/sys/cargo_toml.rs should be the correct file to look at.

**Comment by bilelmoussaoui** [comment]
Can this be closed now?
